app/src/main/java/com/example/composetemplate/presentation/screens/main_screens/MainViewModel.kt

```diff
@@ -2,14 +2,23 @@ package com.example.composetemplate.presentation.screens.main_screens
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.example.composetemplate.MainApplication
 import com.example.composetemplate.data.models.local_models.Test
 import com.example.composetemplate.repositories.TestRepository
+import com.example.composetemplate.utils.Constants.Companion.DS_TEST_KEY
 import com.example.composetemplate.utils.LogsManager
+import com.example.composetemplate.utils.enums.DataStoreType
+import com.example.composetemplate.utils.extensions.readValue
+import com.example.composetemplate.utils.extensions.writeValue
 import com.example.composetemplate.utils.tag
 import com.example.composetemplate.utils.type
 import kotlinx.coroutines.launch
 
-class MainViewModel(private val repository: TestRepository) : ViewModel() {
+class MainViewModel(
+    private val repository: TestRepository,
+    private val application: MainApplication
```

Comment:
why we need to inject application class to viewModel? viewModel should get data layers no application. you can consider inject context here or context as a paramter, we can talk about it if you want

Reply:
The Application is actually a subclass of Context, so basically, it's just like injecting a context. I just need a context to run the DS code. If you have a better idea, I would be glad to hear it.

Reply:
Actually I have a better Idea. instead of injecting application to viewModels, you can create another class like DataStoreManager that you inject the application to there and set the methods of read and write over there. that way you can inject to the view model the DataStoreManager instead of application and set the all logic in DataStoreManager that related to DS, instead do DS login in application